quizzes/quiz3.md

```diff
@@ -0,0 +1,221 @@
+```
+
+So did you notice that in one case we used the - character to separate multiwords (**background-color**) and in the other case we used the camel case (**backgroundColor**)
+
+We are going to learn about the object syntax in the coming lessons.
+
+___
+
+**document variable**
+
+document is an object that web browser supplies to us for changing the HTML document. If you want to access any element in an HTML document, you always start with accessing the document object.
+
+___
+
+**accessing an element if you know its id**
+
+If in your html file you have created an id attribute for an html element, you can access this element in JavaScript using the document.getElementById() method. (NEW INFO: method is a function that belongs to an object) For example in your html if you have:
+
+```html
+<input id="age" type="number"/>
+```
+
+Then in JavaScript you can access it using:
+
+```js
+document.getElementById("age");
+```
+
+Remember JavaScript is quite picky about the case of the id attribute's value. So you can't say:
+
+```js
+document.getElementById("AGE");
+```
+
+If you do, then you get a **null** back.
+
+There are **multiple ways** of accessing an HTML element in DOM language, good news is that in this course we will be using **only** the `getElementById` method.
+
+#### 2. Name two ways of executing JavaScript code using HTML.
+
+**First way** is to embed the code directly into the HTML file. Like this:
+
+```html
+<html>
+    <body>
+        <script>console.log("hello");
+        </script>
+    </body>
+</html>
+```
+or
+
+```html
+<html>
+    <body>
+        <button onclick="console.log("hello");" />
+    </body>
+<html>
+```
+
+**Second way** is to link the JavaScript file using the src attribute of the script tag. Like this:
+
+```html
+<html>
+    <body>
+        <script src="main.js"></script>
+    </body>
+</html>
+```
+
+#### 3. What is the advantage of linking a separate JavaScript source file into our HTML file instead of writing code directly in the HTML file?
+
+**One reason** is that, HTML uses a different syntax than the JavaScript and for a programmer switching between two different contexts might be difficult.
+
+**Another reason** is that, we can validate an HTML code but we can not execute HTML code step by step. We can put breakpoints into a JavaScript file but we can not put breakpoints into an HTML file. So although technically we can write JavaScript code inside the HTML code, we try to keep this to a minimum.
+
+#### 4. What is the reason that we put `<script>` tag as the last tag in the `<body>` tag?
+
+Because while the browser is executing your HTML file line by line, if it encounters a `<script>` tag, it will start executing the JavaScript code line by line before continuing the execution of your HTML. If you try to access an HTML element which is not defined yet, it will get null back. To prevent this scenario, we write the code associated with the `<script>` tag at the bottom of the `<body>` tag. 
```

Comment:
```suggestion
While the browser is executing your HTML file line by line, if it encounters a `<script>` tag, it will start executing the JavaScript code line by line before continuing the execution of your HTML. If you try to access an HTML element which is not defined yet, it will return `null`. To prevent this scenario, we write the code associated with the `<script>` tag at the bottom of the `<body>` tag. 
```